tracing_test.go

```diff
@@ -138,3 +139,43 @@ func TestTracingSpanAttributes(t *testing.T) {
 		assert.NotNil(t, child.Tag("peer.port"))
 	})
 }
+
+// Per https://github.com/uber/tchannel-go/issues/505, concurrent client calls
+// made with the same shared map used as headers were causing panic due to
+// concurrent writes to the map when injecting tracing headers.
+func TestReusableHeaders(t *testing.T) {
+	tracer := mocktracer.New()
+
+	opts := &testutils.ChannelOpts{
+		ChannelOptions: ChannelOptions{Tracer: tracer},
+		DisableRelay:   true,
+	}
+	WithVerifiedServer(t, opts, func(ch *Channel, hostPort string) {
+		jsonHandler := &JSONHandler{TraceHandler: testtracing.TraceHandler{Ch: ch}, t: t}
+		json.Register(ch, json.Handlers{"call": jsonHandler.callJSON}, jsonHandler.onError)
+
+		span := ch.Tracer().StartSpan("client")
+		ctx := opentracing.ContextWithSpan(context.Background(), span)
```

Comment:
I'll add validation, otherwise we don't know if tracing is even propagated / headers being modified